main/models.py

```diff
@@ -123,10 +127,51 @@ class Reimbursement(models.Model):
 
     def to_dict(self):
         return {
+            "id": self.id,
             "user": self.user.username,
             "amount": self.amount,
             "reason": self.reason,
             "status": self.status,
             "type": self.type,
             "expense_date": self.expense_date,
         }
+
+class Payroll(models.Model):
+    profile = models.OneToOneField(Profile, on_delete=models.CASCADE, related_name="payroll", null=True)
+    basic_salary = models.DecimalField(max_digits=10, decimal_places=2)
+    hra = models.DecimalField(max_digits=10, decimal_places=2, default=0.0)
+    special_allowance = models.DecimalField(max_digits=10, decimal_places=2, default=0.0)
+    tax_deduction = models.DecimalField(max_digits=10, decimal_places=2, default=0.0)
+    reimbursement_amount = models.DecimalField(max_digits=10, decimal_places=2, default=0.0)
+    final_salary = models.DecimalField(max_digits=10, decimal_places=2, default=0.0)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def calculate_final_salary(self):
+        """
+        Compute the final salary after applying breakups, reimbursements, and deductions.
+        """
+        self.final_salary = (
+            self.basic_salary 
+            + self.hra 
```

Comment:
why ?

Reply:
all the allowances  and reimbursements are added to the salary and tax is deducted. So to get the full split and final salary we have all the fields.

Reply:
check comments in your POC